testsuite/forge/src/backend/k8s/cluster_helper.rs

```diff
@@ -103,6 +104,46 @@ async fn wait_genesis_job(kube_client: &K8sClient, era: &str, kube_namespace: &s
     .await
 }
 
+async fn wait_node_haproxy(
+    kube_client: &K8sClient,
+    kube_namespace: &str,
+    num_haproxy: usize,
+) -> Result<()> {
+    aptos_retrier::retry_async(k8s_wait_nodes_strategy(), || {
+        let deployments_api: Api<Deployment> = Api::namespaced(kube_client.clone(), kube_namespace);
+        Box::pin(async move {
+            for i in 0..num_haproxy {
+                let haproxy_deployment_name =
+                    format!("{}-{}-haproxy", APTOS_NODE_HELM_RELEASE_NAME, i);
```

Comment:
small rust pro tip, you can rewrite this to:
```suggestion
                    format!("{APTOS_NODE_HELM_RELEASE_NAME}-{i}-haproxy");
```
which is slightly more readible